app/assets/javascripts/heatmap.js

```diff
@@ -59,9 +60,12 @@ document.addEventListener("DOMContentLoaded", () => {
       },
       scale: {
         color: {
-          type: "threshold",
+          type: "linear",
+          // reverse mutates the array. But with theme changes we must work on the same data.
           range: currentTheme === "dark" ? rangeColors : Array.from(rangeColors).reverse(),
-          domain: [10, 20, 30, 40]
+          domain: [0, heatmapData.reduce((max, item) => {
+            return item.total_changes > max ? item.total_changes : max;
+          }, 0)]
```

Comment:
I think the 20 % performance loss is worth going for the more concise option:
```suggestion
          domain: [0, Math.max(...heatmapData.map(d => d.total_changes)), 0)]
```

Reply:
I was not was not awake enough to fiddle myself something and trusted copilot to find something smart. :-(
Thanks for your input!
Changed to something similar like your suggestion.

Reply:
Well, it's faster, but we're not exactly doing massive-scale data science here.